force-app/main/default/classes/TestDebuggingTest.cls

```diff
@@ -0,0 +1,629 @@
+        testLeads.add(new Lead(LastName = 'Shmo 4',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 4',  LeadSource = '',    Email = 'Shmo4@gmail.com', Phone = '', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 5',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 5',  LeadSource = '',    Email = '', Phone = '210-254-8377', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 6',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 6',  LeadSource = 'Web', Email = 'Shmo6@gmail.com', Phone = '210-254-8377', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 7',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 7',  LeadSource = 'Web', Email = '', Phone = '210-254-8377', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 8',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 8',  LeadSource = '',    Email = 'Shmo8@gmail.com', Phone = '210-254-8377', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 9',  Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 9',  LeadSource = '',    Email = '', Phone = '210-254-8377', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 10', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 10', LeadSource = 'Web', Email = 'Shmo10@gmail.com', Phone = '210-254-8377', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 11', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 11', LeadSource = 'Web', Email = '', Phone = '210-254-8377', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 12', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 12', LeadSource = '',    Email = 'Shmo12@gmail.com', Phone = '210-254-8377', Industry = 'Healthcare'));
+        testLeads.add(new Lead(LastName = 'Shmo 13', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 13', LeadSource = '',    Email = '', Phone = '', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 14', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 14', LeadSource = 'Web', Email = 'Shmo14@gmail.com', Phone = '', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 15', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 15', LeadSource = 'Web', Email = '', Phone = '', Industry = 'Technology'));
+        testLeads.add(new Lead(LastName = 'Shmo 16', Title = 'Manager', Status = 'Open - Not Contacted', Company = 'Firehouse Subs 16', LeadSource = '',    Email = 'Shmo16@gmail.com', Phone = '', Industry = 'Technology'));
```

Comment:
Try creating a test data factory that can generate the leads for you with a list of leads sources and other values that you are testing